### ISSUE TITLE
Inspector should expose Resource's Script property so scripts can be set in nested resource fields

### ISSUE THREAD

**Issue by blackears** [godotengine/godot-proposals#4606]
### Describe the project you are working on

Card game.

### Describe the problem or limitation you are having in your project

I am programming in C#.  I've created several scripts that extend Resource that I'm using to model my data.  Some of these scripts have fields that link to other resources.  I would like to create instances of these in the inspector and set their properties.  The way to do this with C# scripts is to create an base Resource object by right clicking in the FileSystem panel and selecting New Resource and then filling in the Script field of that resource to be a path to the C# script you compiled.  

This works fine for the base resource, but creates problems when that resource contains a child resource.  For example, I have already created a Resource object and assigned it the script MonsterPartyRes .  I then created as new array for ```monsters``` of size 1.  I now want to add a MonsterRes to the ```monsters``` array.  If I try to do this through the inspector, I can create a new base Resource object, but the inspector provides no way to assign it a script.  

While I could create these objects in the file system instead. for some types of resource data it makes much more sense for them to exist only as internal data of existing resources.  It seems this functionally already exists for GDScript based resources but since the editor does not recognize C# resources, they need to be assigned  explicitly via their Script field.  It would be very helpful to expose this field in the inspector for resources that are fields of other resources.

```
using Godot;
using System;

public class MonsterPartyRes : Resource
{
    [Export] public MonsterRes[] monsters;

}
----
using Godot;
using System;

public class MonsterRes : Resource
{
    [Export] public string name = "";
    [Export] public Vector2 position;
}

```
![image](https://user-images.githubusercontent.com/6724608/171045048-5b84484b-59f2-404b-8c3d-440557d378b4.png)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Will allow C# uses to use nested resources without having to explicitly create them in the file system.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Expose the Script field for Resources in the Inspector.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Requires creating new resources for every child resource you are going to use.  For some types of resource data, this will create hundreds of extra hard to manage files.

### Is there a reason why this should be core and not an add-on in the asset library?

Requires changing Inspector.

**Comment by ironfairy42** [comment]
I was having the same issue, but if your class inherited from Resource has the GlobalClass tag, then it can be created directly from the inspector without creating a new file with as many layers of nested Resources as you want (I've only teste 22 layers of nesting, but it should be plenty).
I believe this issue is solved as of Godot 4.1.1
![image](https://github.com/godotengine/godot-proposals/assets/77514863/c122c7e0-991a-4ed7-b59f-38bbde2b95b4)

**Comment by Calinou** [comment]
Closing per the above comment.
